### ISSUE TITLE
Permission denied for installing precompiled module in EKS (CentOS) node

### ISSUE THREAD

**Issue by resworld** [falcosecurity/falco#624]
Hey guys, 

I am trying to use falco in latest EKS node AMIs that AWS provides for their EKS cluster. First I tried was to run as it was for previous version and I got a `perrmission denied` error when falco tries to install module. It successfully downloaded from s3 bucket, but seems like there is no permission to install module from /root/.sysdig directory:

```
* Setting up /usr/src links from host
* Unloading falco-probe, if present
* Running dkms install for falco
Error! echo
Your kernel headers for kernel 4.14.106-97.85.amzn2.x86_64 cannot be found at
/lib/modules/4.14.106-97.85.amzn2.x86_64/build or /lib/modules/4.14.106-97.85.amzn2.x86_64/source.
* Running dkms build failed, couldn't find /var/lib/dkms/falco/0.15.0/build/make.log
* Trying to load a system falco-probe, if present
* Trying to find precompiled falco-probe for 4.14.106-97.85.amzn2.x86_64
Found kernel config at /host/boot/config-4.14.106-97.85.amzn2.x86_64
* Trying to download precompiled module from https://s3.amazonaws.com/download.draios.com/stable/sysdig-probe-binaries/falco-probe-0.15.0-x86_64-4.14.106-97.85.amzn2.x86_64-a7281d2454de7f0821a9f5b26148085e.ko
Download succeeded, loading module
insmod: ERROR: could not insert module /root/.sysdig/falco-probe-0.15.0-x86_64-4.14.106-97.85.amzn2.x86_64-a7281d2454de7f0821a9f5b26148085e.ko: Permission denied
Sat May 25 19:08:12 2019: Falco initialized with configuration file /etc/falco/falco.yaml
Sat May 25 19:08:12 2019: Configured rules filenames:
Sat May 25 19:08:12 2019:    /etc/falco/falco_rules.yaml
Sat May 25 19:08:12 2019:    /etc/falco/falco_rules.local.yaml
Sat May 25 19:08:12 2019:    /etc/falco/rules-php-fpm.yaml
Sat May 25 19:08:12 2019: Loading rules from file /etc/falco/falco_rules.yaml:
Sat May 25 19:08:12 2019: Loading rules from file /etc/falco/falco_rules.local.yaml:
Sat May 25 19:08:12 2019: Loading rules from file /etc/falco/rules-php-fpm.yaml:
Sat May 25 19:08:13 2019: Unable to load the driver. Exiting.
Sat May 25 19:08:13 2019: Runtime error: error opening device /host/dev/falco0. Make sure you have root credentials and that the falco-probe module is loaded.. Exiting.
```

I tried with to use a dev image, but there problem was that couldn't find precompiled module in s3 bucket, so I guess it's not prepared yet:
```
* Trying to download precompiled module from https://s3.amazonaws.com/download.draios.com/stable/sysdig-probe-binaries/falco-probe-0.1.2611dev-x86_64-4.14.106-97.85.amzn2.x86_64-a7281d2454de7f0821a9f5b26148085e.ko
Download failed, consider compiling your own falco-probe and loading it or getting in touch with the sysdig community
```

Does anybody has an idea what I could try to make it work in amazon's latest EKS AMIs or what I did wrong? I am using latest yaml files for k8s with daemonset and with rbac included.

Any suggestion will be helpful. Thanks.

**Comment by videocursoscloud** [comment]
Same problem here.

**Comment by fntlnz** [comment]
@resworld  I need some time to test this out, I suspect that there's some package missing in that CentOS node. 

Can you try to install the `kernel-devel` package before than starting Falco?

**Comment by resworld** [comment]
Thank you, @fntlnz for your answer.

I tried with your suggestion - installing `kernel-devel` package in node servers and after deploing Falco. This fixed mine issue with compiling module in falco startup script and installing it successfully after. So I may use that solution. It was needed to remove readOnly tag for `/host/lib/modules` volume also.

We've found that problem with `permission denied` was caused by SeLinux in CentOS which was set to Enforcing. When I changed it with permissive - it was enough to insert module in first place. So maybe this is one of the requirements for CentOS nodes. This idea comes from @browseman

Thank you for your help!
